src/markupsafe/__init__.py

```diff
@@ -103,8 +103,10 @@ def __mul__(self, num: int) -> "Markup":
     def __mod__(self, arg: t.Any) -> "Markup":
         if isinstance(arg, tuple):
             arg = tuple(_MarkupEscapeHelper(x, self.escape) for x in arg)
-        else:
+        elif hasattr(type(arg), "__getitem__") and not isinstance(arg, str):
             arg = _MarkupEscapeHelper(arg, self.escape)
+        else:
+            arg = (_MarkupEscapeHelper(arg, self.escape),)
```

Comment:
Maybe have a special case for str arguments e.g. 
```python
[el]if isinstance(arg, str):
    arg = arg
```
?

Reply:
I'm going to roll this part back, I think we should be using the helper consistently, especially since it will be used for string values in a tuple.